### PR TITLE
Fix#1678 check if server is started before changing UI

### LINKED ISSUE #1678: Rotating the screen changes the UI in Host Books
**Describe the bug**
Rotating the host books screen before starting the server messes up the UI.

**Expected behavior**
No change in behavior should happen.

**Steps to reproduce the behavior:**
1. Go to `Host Books`
2. Rotate the screen

**Screenshots**

**Environment**
- OS version : 9.0

### PR DESCRIPTION
**Pull request by iadeelzafar**
Fixes #1678 

If the server is turned on before screen rotation. only then the screen UI will be changed.

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt [hunk 1]
```diff
@@ -103,7 +103,7 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
       bookDelegate,
       BookOnDiskDelegate.LanguageDelegate
     )
-    if (savedInstanceState != null) {
+    if (savedInstanceState != null && ServerUtils.isServerStarted) {
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt]: Is the null check redundant?

Comment (macgills) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt]: the bundle is used to retrieve the ip on the next line, I guess IP could be retrieved from ServerUtils? My memory of this screen has gotten a might hazy from the holidays

Comment (macgills) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt]: Yeah @iadeelzafar I'd say you could just delete this if block and just rely on `onResume` as it is 
```
  override fun onResume() {
    super.onResume()
    presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
    if (ServerUtils.isServerStarted) {
      ip = ServerUtils.getSocketAddress()
      layoutServerStarted()
    }
  }
```

**Comment by kelson42** [comment]
@iadeelzafar Can you please fix this PR? Or should we close it?

**Comment by iadeelzafar** [comment]
I'll make the required changes soon.

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt (hunk 1 above)
Comment (iadeelzafar) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt]: @macgills fixed.

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt [hunk 2]
```diff
@@ -264,13 +261,6 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
     )
   }
 
-  override fun onSaveInstanceState(outState: Bundle) {
-    super.onSaveInstanceState(outState)
-    if (ServerUtils.isServerStarted) {
-      outState!!.putString(ipStateKey, ip)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt]: `ipStateKey` should also be deleted

Comment (iadeelzafar) [reply app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt]: @macgills done

**Comment by macgills** [comment]
Just the merge conflict to resolve now @iadeelzafar

**Comment by kelson42** [comment]
@macgills You mean, @iadeelzafar should rebase? ;)

**Comment by macgills** [comment]
@kelson42 as per our [contribution guidelines](https://github.com/kiwix/kiwix-android/blob/develop/CONTRIBUTING.md#branching)
>All branches should have distinct history and should be visually easy to follow, for this reason only perform merge commits when merging code either by PR or when synchronising.
If you wish to rebase you should be following the Golden Rule and ahere to the advice in the heading Aside: Rebase as cleanup is awesome in the coding lifecycle.

Prefer merging from the target branch when work has already been pushed to the repo and is under review

**Pull request merged**